### DISCUSSION TITLE
cypress write

### DISCUSSION THREAD

**Discussion by stirbul2** [cypress-io/cypress#18142] [Questions and Help]
HELLO,i have this code : 

describe('Collect all products pages', () => {
  beforeEach(() => {
    cy.visit('https://www.robofun.ro/')

    cy.get('li a.title').each(($el, index, $list) => {
      
      categories.push({link: $el.attr('href'), name: $el.text().trim(), products: []})
      
    })
  })

  let categories = []

  it('Collect all products pages', async () => {
    

    for (let i = 0; i < categories.length; i++) {
      const category = categories[i];
      
      cy.visit(category.link);
      cy.get('div.product-listing').find('a.title').then($elements => {
       
        for (let j = 0; j < $elements.length; j++) {
          const element = $elements[j];
          category.products.push(element.getAttribute('href'));

          cy.visit(element.getAttribute('href'))
          cy.get('h1.title').then($el => {
            cy.log($el.text().trim())
          })
          cy.get('.fPrice').then($el => {
            cy.log($el.text().trim())
          })
          cy.get('div.short-description').then($el => {
            cy.log($el.text().trim())
          })
        }
        
      })

    }

   

  })
})

Does anybody know how can i make it write every log(name, stock, price) into a fixture to have them all there after i run it?

**Comment by iamghulamnabi** [comment]
First of all create a json file in fixtures folder for example "Data.json"
then :
`it('Collect all products pages', async () => {

       for (let i = 0; i < categories.length; i++) {
          const category = categories[i];
  
            cy.visit(category.link);
            cy.get('div.product-listing').find('a.title').then($elements => {
   
      for (let j = 0; j < $elements.length; j++) {
           const element = $elements[j];
           category.products.push(element.getAttribute('href'));

           cy.visit(element.getAttribute('href'))
           cy.get('h1.title').then($el => {
           cy.log($el.text().trim())
           let Text1 =$elem.text()
           const filename = "cypress/fixtures/Data.json"
           cy.readFile(filename).then((list) => {
           cy.writeFile(filename, [ {'Title':Text1}])
                })
             })
         cy.get('.fPrice').then($el => {
         cy.log($el.text().trim())
          let Text2 =$elem.text()
           const filename = "cypress/fixtures/Data.json"
           cy.readFile(filename).then((list) => {
            cy.writeFile(filename, [ {'Title':Text2}])
             })
        })
      cy.get('div.short-description').then($el => {
        cy.log($el.text().trim())
         let Text3 =$elem.text()
           const filename = "cypress/fixtures/Data.json"
           cy.readFile(filename).then((list) => {
            cy.writeFile(filename, [ {'Title':Text3}])
           })
        }  
      })
     }
   })
})`
